modules/helfi_node_survey/helfi_node_survey.module

```diff
@@ -58,3 +64,35 @@ function helfi_node_survey_form_node_form_alter(&$form, &$form_state, $form_id)
 function helfi_node_survey_content_translation_status_access($form_id) : bool {
   return !in_array($form_id, ['node_survey_edit_form', 'node_survey_form']);
 }
+
+/**
+ * Implements hook_entity_access().
+ *
+ * Prevent unauthenticated from accessing survey-node page.
+ */
+function helfi_node_survey_entity_access(
+  EntityInterface $entity,
+  $operation,
+  AccountInterface $account,
+): AccessResult {
+  if ($entity->bundle() !== 'survey') {
+    return AccessResult::neutral();
+  }
+
+  if ($account->isAnonymous()) {
+    return AccessResult::forbidden();
```

Comment:
I think this breaks global surveys & announcements, since anonymous users no longer have view permissions to the entity.

If you do this, you need to add credentials to the json:api request: https://github.com/City-of-Helsinki/drupal-helfi-platform-config/blob/a0fa7e4d52532b103a826b9b50d965cd7befb304/modules/helfi_etusivu_entities/src/Plugin/ExternalEntities/StorageClient/EtusivuJsonApiEntityBase.php#L175